impl/httpMetricsManager.go

```diff
@@ -2,24 +2,62 @@ package impl
 
 import (
 	"fmt"
+	"sort"
+	"strings"
 	"time"
 
+	"github.com/satori/go.uuid"
 	"github.com/xiaoyao1991/chukonu/core"
 )
 
 type HttpMetricsManager struct {
 	requestSent int
 	countQueue  chan int
+	records     HttpRecords
+	durationSum DurationSummation
+	numOfError  int
 	// throughputRecord []int
 }
 
+type HttpRecord struct {
```

Comment:
yes, but the same data is then present in two places, not memory efficient. I had those getter methods in `ChukonuResponse` because I was thinking to have `ChukonuResponse` to behave similarly as `HttpRecord`